### PR TITLE
chore: remove JS types checker, rely on typescript

### PR DESCRIPTION
**Pull request by dgozman**
We keep checking that all methods are documented, and no extra methods
are documented, but rely on typescript for everything else.

### CODE REVIEW

File: docs-src/api-body.md [hunk 1]
```diff
@@ -3811,9 +3811,6 @@ When all steps combined have not finished during the specified [`option: timeout
 
 Returns the `node.textContent`.
 
-## method: ElementHandle.toString
```
Comment (JoelEinbinder) [inline docs-src/api-body.md:3814]: 👍

File: utils/doclint/check_public_api/missingDocs.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  });
+  const checker = program.getTypeChecker();
+  const apiClassNames = new Set();
+  const apiMethods = new Map();
+  const apiSource = program.getSourceFiles().find(f => f.fileName === apiFileName);
+
+  /**
+   * @param {ts.Type} type
+   */
+  function signatureForType(type) {
```
Comment (JoelEinbinder) [inline utils/doclint/check_public_api/missingDocs.js:106]: JSBuilder lives!

**Review by JoelEinbinder — APPROVED** [review]

File: utils/doclint/check_public_api/missingDocs.js (hunk 2 above)
Comment (dgozman) [reply utils/doclint/check_public_api/missingDocs.js:106]: I was debating whether to add this check for missing parameters, and decided it only costs us this one JSBuilder function 😄

Comment (JoelEinbinder) [reply utils/doclint/check_public_api/missingDocs.js:106]: The typescript compiler api isn't quite stable, so this function has actually broken once or twice. But I think we can afford it.

**Pull request merged**